components/brave_extension/extension/brave_extension/content.ts

```diff
@@ -23,3 +23,52 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
     }
   }
 })
+
+window.addEventListener('DOMContentLoaded', () => {
+  if (!window.web3) {
+    // For dapp detection.
+    const script =
+      `
+      function __insertWeb3Installed() {
+        if (!window.alreadyInserted) {
+          const meta = document.createElement('meta')
+          meta.name = 'web3-installed'
+          document.head.appendChild(meta)
+          window.alreadyInserted = true
+        }
+      }
+      if (window.web3) {
+        if (!window.web3.currentProvider || !window.web3.currentProvider.isMetaMask) {
+          __insertWeb3Installed()
+        }
+      } else {
+        var oldWeb3 = window.web3
+        Object.defineProperty(window, 'web3', {
+          configurable: true,
+          set: function (val) {
+            __insertWeb3Installed()
+            oldWeb3 = val
+          },
+          get: function () {
+            __insertWeb3Installed()
+            return oldWeb3
+          }
+        })
+      }`
+
+    const scriptEl = document.createElement('script')
+    scriptEl.textContent = script;
+    (document.head || document.documentElement).appendChild(scriptEl)
+
+    setTimeout(function () {
+      const isDapp = document.querySelector('meta[name="web3-installed"]')
+      if (isDapp) {
+        chrome.runtime.sendMessage({
+          type: 'dappAvailable',
+          location: window.location.href
```

Comment:
this is not necessary, the url is already in sender.url

Reply:
location is no longer being used.